### PR TITLE
fix: port range exposing and various bugfixes

### PR DESCRIPTION
**Pull request by m8vago**
- Rework agent version check
- Fix an issue in dagent, where the port ranges weren't exposed.
- Clarify some comments
- Fix metrics port validation
- Add http -> https redirection to the main docker-compose

### CODE REVIEW

File: web/crux/src/main.ts [hunk 1]
```diff
@@ -20,6 +20,7 @@ import PrismaErrorInterceptor from './interceptors/prisma-error-interceptor'
 import prismaBootstrap from './services/prisma.bootstrap'
 import { PRODUCTION } from './shared/const'
 import DyoWsAdapter from './websockets/dyo.ws.adapter'
+import { coerce } from 'semver'
```
Comment (robot9706) [inline web/crux/src/main.ts:23]: Is this used?

Comment (m8vago) [reply web/crux/src/main.ts:23]: Yeah, i have forget that one, the linter thrown an error as well.

**Review by polaroi8d — CHANGES_REQUESTED** [review]
I really need some documentation regarding AGENT_SUPPORTED_MINIMUM_VERSION. We need to clarify how we intend to handle it, what it is, why we are using it, and so on. It's my fault, I didn't add this to the previous PR, sorry.

**Review by polaroi8d — CHANGES_REQUESTED** [review]
Test case: Added a new project with 1 nginx container. Added a portrange from 7000 to 7500 same in internal and external. Added a port expose 7543 same in internal and external. Tried to deploy it and got the following error message: `client.js:1 TypeError: Cannot read properties of null (reading 'enabled')` 


Otherhand console throws a few `next-translate` error about "images" for example: 
`[next-translate] "images:" is missing in current namespace configuration. Try adding "" to the namespace "images". `

**Comment by m8vago** [comment]
> add

It's not your fault, i removed this const from the last PR and added it here. I'll write some docs on it.

File: web/crux/src/domain/validation.ts [hunk 2]
```diff
@@ -334,17 +335,24 @@ const createMetricsPortRule = (ports: ContainerPort[]) => {
...
+const metricsRule = yup.mixed().when(['ports'], ([ports]) => {
+  const portRule = createMetricsPortRule(ports)
+
+  return yup
     .object()
-    .shape({
-      path: yup.string().nullable(),
-      port: createMetricsPortRule(ports),
+    .when({
+      is: (it: Metrics) => it.enabled,
```
Comment (robot9706) [inline web/crux/src/domain/validation.ts:344]: `it` can be null so `it?.enabled` should work here.

File: web/crux-ui/src/validations/container.ts [hunk 3]
```diff
@@ -331,17 +332,24 @@ const createMetricsPortRule = (ports: ContainerPort[]) => {
...
+const metricsRule = yup.mixed().when(['ports'], ([ports]) => {
+  const portRule = createMetricsPortRule(ports)
+
+  return yup
     .object()
-    .shape({
-      path: yup.string().nullable(),
-      port: createMetricsPortRule(ports),
+    .when({
+      is: (it: Metrics) => it.enabled,
```
Comment (robot9706) [inline web/crux-ui/src/validations/container.ts:341]: Same as in crux.

File: DEVELOPMENT.md [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+Crux considers two things to determine if an agent is outdated:
+ - The `AGENT_PROTO_COMPATIBILITY_MINIMUM_VERSION` in the `web/crux/shared/const.ts` file
+ - And the version number in Crux's `package.json`
+
+When the agent version is between `AGENT_PROTO_COMPATIBILITY_MINIMUM_VERSION` and the `package.json`'s version
+the agent's status will be connected, and it will be fully functional. Otherwise it going to be outdated,
+and can not be used unless an update is performed.
+
+### Releasing a new agent
+While creating a new release, the `AGENT_PROTO_COMPATIBILITY_MINIMUM_VERSION` should be incremented to the version number of the new release when:
```
Comment (polaroi8d) [inline DEVELOPMENT.md:16]: Based on this, could you please extend the `release` target command in the root `Makefile`, please?

**Review by polaroi8d — APPROVED** [review]

**Review by robot9706 — APPROVED** [review]
Tested locally, ranges and validation seem to work fine.

**Pull request merged**